### PR TITLE
Improved options, help, and auto-complete

### PR DESCRIPTION
**Pull request by americanhanko**
Backwards-incompatible API change due to renaming some options

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ Using [fisher](https://github.com/jorgebucaran/fisher)
 fisher install jorgebucaran/getopts.fish americanhanko/fish-spin
 ```
 
+## Dependencies
+
+- Ruby
+- Python 3
+- `jorgebucaran/getopts.fish`
```
Comment (jorgebucaran) [inline README.md:18]: Maybe we can get rid of this dependency later.

**Review by jorgebucaran — APPROVED** [review]
Looks good! I'll send you another PR with some stylistic changes. Thank you for taking the project, it was too much for me. 💯

**Comment by publicarray** [comment]
Can I ask why break backwards compatibility? I get the new options are cleaner but could this break existing scripts that rely on this?

**Comment by jorgebucaran** [comment]
Probably not many dependents (if at all) right now.

**Comment by publicarray** [comment]
I've only found 1 or 2 (if you count https://github.com/sijad/gitignore/ I'd like to add it back in)

https://github.com/dideler/fish-cd-git uses `spin --format $spinner_msg $clone_cmd` so we are good :+1: 

GH Search: https://github.com/search?q=jorgebucaran%2Ffish-spin&type=code

**Pull request merged**